File: java/src/test/java/com/hakanozbay/metric/EditDistanceTest.java
package com.hakanozbay.metric;

import static org.junit.Assert.assertEquals;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

public class EditDistanceTest {

	EditDistance editDistance;
	
	@Before
	public void setUp() 
	{
		editDistance = new EditDistance();
	}

	@After
	public void tearDown() 
	{
		editDistance = null;
	}

	@Test
	public void testEditDistanceOfIdenticalWords()
	{
		double distance = editDistance.calculate("hi", "hi");
		assertEquals(0, distance, 0);
	}
	
	@Test
	public void testEditDistanceOfIdenticalWordsCaseSensitive()
	{
		double distance = editDistance.calculate("hi", "Hi");
		assertEquals(1, distance, 0);
	}
	
	@Test
	public void testEditDistanceOfSimilarWords()
	{
		double distance = editDistance.calculate("cat", "bat");
		assertEquals(1, distance, 0);
	}
	
	@Test
	public void testEditDistanceOfSubsetWords()
	{
		double distance = editDistance.calculate("cat", "catch");
		assertEquals(2, distance, 0);
	}
	
	@Test
	public void testEditDistanceOfAnagrams()
	{
		double distance = editDistance.calculate("tame", "mate");
		assertEquals(2, distance, 0);
	}
	
	@Test
	public void testEditDistanceOfWordsWithSimilarLetters()
	{
		double distance = editDistance.calculate("act", "catch");
		assertEquals(3, distance, 0);
	}
	
}

File: python/test/editDistanceTest.py
import unittest
from editDistance import calculate

class EditDistanceTest(unittest.TestCase):

	def test_edit_distance_of_identical_words(self):
		editDistance = calculate("hi","hi")
		self.assertEqual(editDistance, 0)

	def test_edit_distance_of_identical_words_case_senseitive(self):
		editDistance = calculate("hi","Hi")
		self.assertEqual(editDistance, 1)		

	def test_edit_distance_of_similar_words(self):
		editDistance = calculate("cat","bat")
		self.assertEqual(editDistance, 1)		

	def test_edit_distance_of_subset_words(self):
		editDistance = calculate("cat","catch")
		self.assertEqual(editDistance, 2)

	def test_edit_distance_of_anagrams(self):
		editDistance = calculate("tame","mate")
		self.assertEqual(editDistance, 2)

	def test_edit_distance_of_words_with_similar_letters(self):
		editDistance = calculate("act","catch")
		self.assertEqual(editDistance, 3)
